### DISCUSSION TITLE
Libre Computer Board AML-S905X-CC (Le Potato) Works!

### DISCUSSION THREAD

**Discussion by goobius** [mcguirepr89/BirdNET-Pi#642] [Show and tell]
Great news! I have successfully installed Birdnet Pi onto a Libre Computer Board AML-S905X-CC (Le Potato)!
It takes a few more steps, but much cheaper than current Raspberry Pi boards, IF you can find one. 
What you need:
•	A working Raspberry Pi (borrowed from a friend)
•	MicroSD card
•	Rasbian 64bit OS
•	Imager (Raspberry Pi Imager)
•	Libre Computer Board ([Amazon link](https://www.amazon.com/Libre-Computer-AML-S905X-CC-Potato-64-bit/dp/B074P6BNGZ/ref=sxts_rp_s_1_0?content-id=amzn1.sym.154791ce-888e-46c9-af88-2c4c0ceada82%3Aamzn1.sym.154791ce-888e-46c9-af88-2c4c0ceada82&crid=100H6VVDIZWX2&cv_ct_cx=le+potato&keywords=le+potato&pd_rd_i=B074P6BNGZ&pd_rd_r=911ce626-3df5-4517-a44c-e4b89ac122e8&pd_rd_w=l9Fr8&pd_rd_wg=Oykvn&pf_rd_p=154791ce-888e-46c9-af88-2c4c0ceada82&pf_rd_r=XRD0FJFG8PN0V99CQR4B&psc=1&qid=1667566630&qu=eyJxc2MiOiIwLjA2IiwicXNhIjoiMC4wNyIsInFzcCI6IjAuMDAifQ%3D%3D&sprefix=le+potato%2Caps%2C104&sr=1-1-5e1b2986-06e6-4004-a85e-73bfa3ee44fe))
•	Power supply
•	USB sound card
•	Microphone (soldered up electret mic onto an aux cable)
Github link: https://github.com/libre-computer-project/libretech-raspbian-portability

1.	Image your RPi, add local account, and fully update and upgrade.
2.	Do not turn on VNC yet. I did, but after the port over to the Libre board, VNC Server was broken. 
3.	Go to the github page for the Libre Computer Raspbian Portability tool. Follow the terminal instructions. It is automated and will prompt you to unplug your RPi after it is done. 
4.	Transfer the MicroSD card to your Libre Computer board. 
5.	Turn on and follow the instructions. 
6.	Set up Birdnet Pi and install the USB sound card plus mic
The Libre Computer board is functioning well and is currently $40 on Amazon! I have added screenshots of the boards stats while analyzing. 
The board has comparable specs to a Raspberry Pi, but lacks onboard WiFi and Bluetooth. I am using mine on a wired network, so it isn’t an issue. I have read that USB WiFi dongles work well with the board. There are GPIO connections, but you have to map them with a wiretools app. There is no plan to connect anything to the GPIO pins, so I haven’t tested that option. 

I will be ordering the heatsink from Amazon to get the temps down a bit, but the board will always be in an air conditioned room. 

Anyone else messing with Le Potato?

<img width="721" alt="hawk" src="https://user-images.githubusercontent.com/30322148/199977654-815ce176-b582-47fc-938f-9409160628c6.png">
<img width="481" alt="SystemInfo" src="https://user-images.githubusercontent.com/30322148/199977718-bc4812d7-6de0-4156-bde0-2fdd3b84f053.png">

**Comment by jflyons** [comment]
Thanks for information. I am going to give this a try :)

**Reply by goobius** [reply]
Sure! Given the prices of RPi's on the open market, you buy two or three of these!

**Reply by rottenoasis** [reply]
I got this to work as well on the same board for a week now! Using with a cheap USB lava mic on Amazon. I have the blue heat sink inside a 3d printed case and no heat problems.  USB wifi dongle works too. It has same footprint and plug layout as raspberry pi 3b+ so you can use those cases

**Reply by goobius** [reply]
Awesome! I am waiting on the heatsink to arrive to lower my current heat values. Glad the USB WiFi dongle worked!

**Reply by bcardi0427** [reply]
I'm thinking of getting a Libre Renegade with 4G Ram and trying it. What type of mic and wifi dongle did you use?

**Reply by goobius** [reply]
I am using a Sabrent USB sound card and a basic electret microphone. My next option is an amplified/filtered microphone module, but I haven't tested it yet. 

The Sabrent USB sound card has been really useful in my Amateur radio setup as well  and doesn't require drivers.

**Reply by bcardi0427** [reply]
I'm not sure how good they are but I found this one that gives you three usb cards for $6.88. I think most of these inexpensive usb cards are pretty much the same. It's Amazon, I can always return it. 

https://www.amazon.com/gp/product/B09VBWJQBV/ref=ox_sc_act_title_6?smid=A1ZWNP05VR80QB&psc=1

**Reply by bcardi0427** [reply]
I've got my Le Potato set up and running BirdNet just fine and didn't have to use the portability. I downloaded Libre's distro's https://distro.libre.computer/ci/raspbian/11/ and then did curl -s https://raw.githubusercontent.com/mcguirepr89/BirdNET-Pi/main/newinstaller.sh | bash and it magically ran just fine on ethernet (my usb wifi hub is coming tomorrow). This was easy and I used some small heatsinks (bought 100 assorted sizes for $10 on Amazon) and it feels to the hand that this thing is not heating as much as a pi. I'm using an old usb soundblaster play3 and a cheap mic and it's picking up just fine. Once I get an ip for it I'll put it up here so others can see. I am on Birdweather so that was easy to set up. Great programing on BirdNET-Pi as everything is soooo easy to set up!

**Reply by jmtuts** [reply]
bcardi0427 - Did you get your WiFi hub working?  I have mine running on ethernet but need to to be wifi.  If you have any tips it would be appreciated.

**Reply by bcardi0427** [reply]
Yes I got it working very good with a Panda 150, it was plug and play. https://www.amazon.com/dp/B00762YNMG?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Reply by jmtuts** [reply]
I am up and running with the Libre AML-S905X-CC (Le Potato).  A big thank to all who provided info and guidance.  No more pops or snaps, just clean data.  Same power supply BTW, so I suspect my Rpi was the issue, it has been repurposed to ADS-B work for FlightAware

**Reply by bcardi0427** [reply]
I also found that using the new model (3.5K) is far better than the 6K one. On the 6K one I was slowing down and had to reboot about halfway through the day but the 3K one never seems to slow down and temps stay lower.

**Reply by bcardi0427** [reply]
I'm getting about 400-800 detections a day with the new model and was getting 200-400 on the other model.

**Comment by jmtuts** [comment]
So, no config, it just worked?  Sounds too goo to be true ;-)

**Comment by arduionoGP** [comment]
It also works on the Renegade board. You can download the adjusted Bullseye OS directly from the Libre distro download server without having to use the Portability utility and then just directly install birdnetpi using the link on the site. It just works - both potatoe and renegade using the specific distros on

**Reply by blogdemoi** [reply]
I think I am going to buy one Le Potato to replace my old raspberry Pi 3B.

**Reply by jmtuts** [reply]
I did, and couldn't he happier.  The 3B and 3B+ have issues, going to the Potato solved many problems I was having.  Plus, you can't beat the price.

**Reply by arduionoGP** [reply]
The Renegade board is quite a bit faster and not much more in price. It also can be had with 4gb ram. Both my potato and renegade have been running birdnetpi just fine.

**Reply by blogdemoi** [reply]
> I did, and couldn't he happier. The 3B and 3B+ have issues, going to the Potato solved many problems I was having. Plus, you can't beat the price.

I will order it from the US, it is still cheaper than buying it from France. Do you use the 1GB or 2GB version?

**Reply by jmtuts** [reply]
I have the 2gb version and have not seen any performance problems, check out my installation at FallenBirdNet.sytes.net
Marana, Arizona, USA

**Reply by bcardi0427** [reply]
I'm looking for a board that will run Raspbian like the LePotato but has an M.2 slot for a SSD. I'm just a bit worried my LePotato will kill the SD card after a bit and I want this running a long time. If I could I'd love to run it on my x86 server with a VM. Does anyone know how long the SD Cards will last running BirdNetPi? It seems to do a lot of disk writing according to the logs.

See my site at https://birdnetpi.bcardi.org

**Reply by blogdemoi** [reply]
> I have the 2gb version and have not seen any performance problems, check out my installation at FallenBirdNet.sytes.net Marana, Arizona, USA

Thank you. I will order the 2gb version then. Nice to hear your birds!

**Comment by arduionoGP** [comment]
Amazon.

**Comment by arduionoGP** [comment]
4gb

**Comment by bcardi0427** [comment]
Has anyone tried James Chamber's USB SSD mod? https://jamesachambers.com/libre-le-potato-ssd-boot-guide/  I'm thinking of picking up a Renegade and I already have a couple SSD drives and a usb adapter. I'll see if BirdNETPi will work on that, I also want something to run Caddy from and it all might just tie together.

**Comment by blogdemoi** [comment]
Hi,

I have just received my Lepotato, downloaded the .img  [2022-09-22-raspbian-bullseye-arm64-lite+aml-s905x-cc.img.xz](https://distro.libre.computer/ci/raspbian/11/2022-09-22-raspbian-bullseye-arm64-lite%2Baml-s905x-cc.img.xz ) file from https://distro.libre.computer/ci/raspbian/11/ , burnt the image using the Rpi imager (setting ssh, user, ssid, pswd, etc.) booted the Lepotato.

I attached a usb wifi dongle. Unfortunately the Lepotato doesn't appear on my wifi network.

Do I have to install any driver for a usb wifi dongle? Am I missing something?

Thank you.

**Comment by arduionoGP** [comment]
Some USB dongles work automatically, like the one libre packages with their
board on amazon. Some don't without messing with drivers.

**Reply by blogdemoi** [reply]
Finally I got it working. It seems that when flashing the sd card with the official RPI Imager, entering the right infos to enable ssh and ssid/psswd, didn't work, I don't know why.

So I manually added a ssh file, wpa_supplicant.conf and userconf files in the boot partition (sd card plugged in another Linux computer), plugged in my wifi dongle and booted  and my Lepotato is on my wifi network. I am currently installing Birdnet-Pi now on it.

**Comment by arduionoGP** [comment]
Excellent. Some USB dongles will work out of the box with potatoe but not
with renegade also BTW. Like the Panda model pau03. Libre seems to package
and support "Gen basic WiFi USB nano wireless network dongle for Linux"
which is running birdnet pi for me on both a renegade and a potatoe Board.

**Comment by blogdemoi** [comment]
Installation complete on my Lepotato. However when on the "System Info" page I have a warning sign with errors (no idea what it is about):

> Errors
> 
>     /usr/local/bin/extra_info.sh - sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     sudo: vcgencmd: command not found
>     Return value: 0
> 
>     ./xml.php on line 39
>     ./includes/output/class.WebpageXML.inc.php on line 254 in function getJsonString()
>     ./includes/output/class.WebpageXML.inc.php on line 241 in function getXMLString()
>     ./includes/xml/class.XML.inc.php on line 914 in function getXml()
>     ./includes/xml/class.XML.inc.php on line 903 in function _buildXml()
>     ./includes/xml/class.XML.inc.php on line 941 in function _buildPlugins()
>     ./plugins/viewer/class.viewer.inc.php on line 63 in function execute()
>     ./includes/class.CommonFunctions.inc.php on line 382 in function executeProgram( "extra_info.sh", "", "........................................IPs.....................................\nLAN IP: 10.0.242.192\nPublic IP: 79.143.250.120\n..................................`vcgencmd stats`..............................\nBinary: \n....................................Clock Speeds................................\narm:\t\ncore:\t\nh264:\t\nisp:\t\nv3d:\t\nuart:\t\npwm:\t\nemmc:\t\npixel:\t\nvec:\t\nhdmi:\t\ndpi:\t\n........................................Volts...................................\ncore:\t\nsdram_c:\t\nsdram_i:\t\nsdram_p:\t\n.....................................Caddyfile..................................\nhttp:// {\n root * /home/arnauld/BirdSongs/Extracted\n file_server browse\n handle /By_Date/* {\n file_server browse\n }\n handle /Charts/* {\n file_server browse\n }\n reverse_proxy /stream localhost:8000\n php_fastcgi unix//run/php/php7.4-fpm.sock\n reverse_proxy /log* localhost:8080\n reverse_proxy /stats* localhost:8501\n reverse_proxy /terminal* localhost:8888\n}\n.................................... Crontab....................................\n\nSHELL=/bin/sh\nPATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin\n\n17 *\t* * *\troot cd / && run-parts --report /etc/cron.hourly\n25 6\t* * *\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.daily )\n47 6\t* * 7\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.weekly )\n52 6\t1 * *\troot\ttest -x /usr/sbin/anacron || ( cd / && run-parts --report /etc/cron.monthly )\n*/5 * * * * arnauld /usr/local/bin/disk_check.sh >/dev/null 2>&1\n*/3 * * * * arnauld /usr/local/bin/cleanup.sh >/dev/null 2>&1\n@reboot arnauld /usr/local/bin/cleanup.sh >/dev/null 2>&1\n0 1 * * 1 arnauld /usr/local/bin/weekly_report.sh >/dev/null 2>&1" )

**Comment by bcardi0427** [comment]
I've had that since I started with it and it doesn't seem to matter. The
only place I see that is the Sysinfo command in tools.

Jerry
birdnetpi.bcardi.org

**Reply by blogdemoi** [reply]
So, after a few hours of running the Lepotato I am quite happy with it. Much faster than my old Pi 3B, setting it is easy, all good: https://birdnetpi.arnauld.org/,  my Internet connection is slow, I just have an old ADSL Internet. I am using the new Birdnet-Pi model.

**Reply by blogdemoi** [reply]
Well, almost fine, my "No detections for today" issue is back...

![Screenshot_20230309-150819](https://user-images.githubusercontent.com/83965802/224050590-d5076f9c-9214-402f-a921-384ee12cbb12.png)

**Reply by blogdemoi** [reply]
It's back to normal, the "No detections for Today" issue seems to happen only for a short time, maybe when Lepotato is too busy...

**Reply by blogdemoi** [reply]
I have again my "No detections for Today" issue. As it is a fresh install I wonder what can cause this issue. I have free RAM, free swap...

**Comment by arduionoGP** [comment]
Try playing a cardinal sound directly into the mic with your cell phone
while watching the terminal window messages and make sure all the services
are running from the gui.

**Reply by blogdemoi** [reply]
Now it is working fine.  I guess it is related with the Lepotato being overwhelmed sometimes... But I will try to follow your suggestion when I have the "No detections for Today" issue again.

**Comment by bcardi0427** [comment]
For a bit I had some heating problems on my LePotato and it would take it
hours to catch up. Look at the date and time in the logs of what is now
going on and see if it's behind real time. I used to have to restart the
analyzer and that would help. I finally resolved that with a new fan.

Bcardi
https://birdnetpi.bcardi.org

**Comment by croisez** [comment]
By experience, how would you compare LePatato to Raspberry-Pi ?
Is it comparable to RPI3, or RPI4 ?

**Comment by arduionoGP** [comment]
Running birdnet it feels a bit faster than a B+, the Libre Renegade feels a
lot faster.

**Reply by croisez** [reply]
By B+, you mean a RPI3-B+ ? Or a RPI4

**Comment by arduionoGP** [comment]
3-B+

**Reply by croisez** [reply]
ah ok.
I currently run birdnet-pi on a PI3-B+, and I experiment slow responses.
If it is the same with LePotato or Renegade, I don't see any advantages to give-up with my PI3-B+, then

**Reply by blogdemoi** [reply]
I am running a LePotato to replace a Pi 3B. It is faster than a Pi 3B but if you have a Pi 3B+ I think you should keep the Pi 3B+ as I don't think the LePoatato is better the than the Pi 3B+.

**Comment by arduionoGP** [comment]
My B+ would occasionally hang and get behind processing. My potatoe does
not. I like the renegade better because the birdnet web page is snappier
when also doing processing in the background. The only possible downside of
the renegade is it natively supports less USB WiFi sticks so use the
suggested ones by the company.

**Comment by gerritzen** [comment]
Just to add my experience: I transitioned from a raspberry pi 4 to an AML-S905X-CC (Le Potato). It worked flawlessly with the portability tool, wifi with the included USB card and everything. So far, I encountered no problems with performance, compatibility or otherwise.
